src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3679,6 +3679,74 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         return nullptr;
     }
 
+    ASR::asr_t* handle_intrinsic_str(Allocator &al, Vec<ASR::call_arg_t> args,
+                                        const Location &loc) {
+        ASR::expr_t *arg = nullptr;
+        ASR::expr_t *res_value = nullptr;
+        ASR::ttype_t *arg_type = nullptr;
+        if (args.size() > 0) {
+            arg = args[0].m_value;
+            arg_type = ASRUtils::expr_type(arg);
+        }
+        ASR::ttype_t *res_type = ASRUtils::TYPE(ASR::make_Character_t(al, loc,
+                                    1, 1, nullptr, nullptr , 0));
+        if (!arg) {
+            return ASR::make_StringConstant_t(al, loc, "", res_type);
+        }
+        if (ASRUtils::is_real(*arg_type)) {
+            if (ASRUtils::expr_value(arg) != nullptr) {
+                double ival = ASR::down_cast<ASR::RealConstant_t>(
+                                        ASRUtils::expr_value(arg))->m_r;
+                // stringstream for exponential notation formatting.
+                std::stringstream sm;
+                sm << ival;
+                std::string value_str = sm.str();
+                sm.clear();
+                res_value =  ASR::down_cast<ASR::expr_t>(ASR::make_StringConstant_t(al,
+                                loc, s2c(al, value_str), res_type));
+            }
+            return (ASR::asr_t *)ASR::down_cast<ASR::expr_t>(ASR::make_Cast_t(
+                al, loc, arg, ASR::cast_kindType::RealToCharacter,
+                res_type, res_value));
```

Comment:
This will work, right?
```suggestion
            return ASR::make_Cast_t(al, loc, arg, ASR::cast_kindType::RealToCharacter,
                res_type, res_value);
```

Reply:
Done!